custom_components/ha_hatch/light.py

```diff
@@ -61,17 +66,27 @@ def turn_on(self, **kwargs):
             # If 100 is sent to Abode, response is 99 causing an error
             brightness = round(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0)
         else:
-            brightness = self._attr_brightness
+            if isinstance(self.rest_device, RestIot):
+                brightness = 0
+            else:
+                brightness = self._attr_brightness
         if ATTR_RGB_COLOR in kwargs:
             rgb = kwargs[ATTR_RGB_COLOR]
         else:
             rgb = self._attr_rgb_color
 
         _LOGGER.debug(f"turning on light to {rgb} with {brightness}")
-        self.rest_device.set_color(rgb[0], rgb[1], rgb[2], brightness)
+        if isinstance(self.rest_device, RestIot):
+            self.rest_device.set_color(rgb[0], rgb[1], rgb[2], 0, brightness, True)
```

Comment:
I don't think this logic will hold up, the rest_devices are thrown away when updating expired auth so any state you want to preserve should be handled in this repo as the api objects are not persistent

Reply:
Hmm I didn't realize this. So you are saying the cached light values in the library won't exist anymore because that object gets replaced? Where is that expired auth logic at? How is HA made aware of the new object?

Reply:
RestEntity, replace_rest_device could be overridden here if you want to preserve it during replacement. it isn't the easiest code to follow but is invoked in __init__.py setup_connection method

Reply:
I see, okay I'll work on this.

Reply:
I believe this should be more clear now having split it in to multiple objects